flexmeasures/data/models/planning/utils.py

```diff
@@ -110,16 +112,41 @@ def get_prices(
             first_event_start = price_df.first_valid_index()
             last_event_end = price_df.last_valid_index() + resolution
             current_app.logger.warning(
-                f"Prices partially unknown for planning window. Trimming planning window (from {query_window[0]} until {query_window[-1]}) to {first_event_start} until {last_event_end}."
+                f"Prices partially unknown for planning window (sensor {price_sensor.id}). Trimming planning window (from {query_window[0]} until {query_window[-1]}) to {first_event_start} until {last_event_end}."
             )
             query_window = (first_event_start, last_event_end)
         else:
             raise UnknownPricesException(
-                "Prices partially unknown for planning window."
+                f"Prices partially unknown for planning window (sensor {price_sensor.id})."
             )
     return price_df, query_window
 
 
+def inflexible_device_forecasts(
+    query_window: Tuple[datetime, datetime],
+    resolution: timedelta,
+    sensor: Sensor,
+):
+    """
+    :returns: power forecasts (consumption is positive, production is negative)
```

Comment:
Does this still have to do with USEF describing this the other way around than we do? Why do we make this switch here?

Reply:
Both the API and the scheduler use positive power values to denote consumption. The database uses positive power values to denote production. The historical reasons are that we had USEF in mind when designing the API and scheduler, and we took a system operator perspective when designing the database models.